node-installer/node-installer_test.go

```diff
@@ -71,6 +81,44 @@ func TestPatchContainerdConfig(t *testing.T) {
 			configData, err := os.ReadFile(configPath)
 			require.NoError(err)
 			assert.Equal(string(tc.expected), string(configData))
+
+			if tc.expectedTemplate != nil {
+				// Unlike patchContainerdConfig, patchContainerdConfigTemplate
+				// requires the file to exist already. Create one.
+				configTemplatePath := filepath.Join(tmpDir, "config.toml.tmpl")
+				rawConfig, err := toml.Marshal(constants.ContainerdBaseConfig())
+				require.NoError(err)
+				err = os.WriteFile(configTemplatePath, rawConfig, os.ModePerm)
+				require.NoError(err)
```

Comment:
This isn't using an actual template as base, maybe we can embed a template the test can run on?